internal/manifests/collector/servicemonitor.go

```diff
@@ -51,5 +59,40 @@ func ServiceMonitor(cfg config.Config, logger logr.Logger, otelcol v1alpha1.Open
 				},
 			},
 		},
-	}, nil
+	}
+
+	endpoints := []monitoringv1.Endpoint{
+		{
+			Port: "monitoring",
+		},
+	}
+
+	sm.Spec.Endpoints = append(endpoints, endpointsFromConfig(logger, otelcol)...)
+	return &sm, nil
+}
+
+func endpointsFromConfig(logger logr.Logger, otelcol v1alpha1.OpenTelemetryCollector) []monitoringv1.Endpoint {
+	c, err := adapters.ConfigFromString(otelcol.Spec.Config)
+	if err != nil {
+		logger.V(2).Error(err, "Error while parsing the configuration")
+		return []monitoringv1.Endpoint{}
+	}
+
+	exporterPorts, err := adapters.ConfigToExporterPorts(logger, c)
+	if err != nil {
+		logger.Error(err, "couldn't build service monitors from configuration")
+		return []monitoringv1.Endpoint{}
+	}
+
+	endpoints := []monitoringv1.Endpoint{}
+
+	for _, port := range exporterPorts {
+		if strings.Contains(port.Name, "prometheus") {
```

Comment:
shouldn't we only be receiving known exporters by this point? do we need this check?

Reply:
If somebody adds support for new exporter receivers, they will not be filtered here and we will be creating ServiceMonitors for non-desired exporters.

Reply:
maybe when you refactor the interface, we should also have a map that parsers can register themselves with for valid servicemonitor ports, though maybe that's overengineering haha